file/file_prefetch_buffer.cc

```diff
@@ -653,6 +654,11 @@ bool FilePrefetchBuffer::TryReadFromCacheUntracked(
       if (for_compaction) {
         s = Prefetch(opts, reader, offset, std::max(n, readahead_size_));
       } else {
+        if (IsOffsetInBuffer(offset, curr_)) {
```

Comment:
Shouldn't this be called after `Prefetch` call to include bytes not present in buffer and will go for prefetching?

Reply:
My thinking is any bytes we read for the block being requested would have been read from storage anyway, prefetch or no prefetch. So I'm only counting bytes consumed from data already in the prefetch buffer. Another possibility is looking at how many prefetch bytes were not useful because they were skipped. But that seems a bit more complicated.

Reply:
Oh ok! Makes sense.